From: Dominic Curran <dominic.curran@citrix.com>
Subject: mlx4: Add function from kernel_addons directory to fix build

This function is found in:
 mlnx_en-1.5.1.3/SRPMS/mlnx_en-1.5.1.3/kernel_addons/backport/2.6.27_sles11/include/workqueue.h

It is needed to build 1.5.1.3.
It was added at the top of en_netdev.c b/c thats the only place it is used.

diff -r 67f445715f32 drivers/net/mlx4/en_netdev.c
--- a/drivers/net/mlx4/en_netdev.c	Tue Aug 03 11:15:20 2010 +0100
+++ b/drivers/net/mlx4/en_netdev.c	Tue Aug 03 11:15:46 2010 +0100
@@ -35,6 +35,7 @@
 #include <linux/tcp.h>
 #include <linux/if_vlan.h>
 #include <linux/delay.h>
+#include <linux/workqueue.h>
 
 #include <linux/mlx4/driver.h>
 #include <linux/mlx4/device.h>
@@ -44,6 +45,10 @@
 #include "mlx4_en.h"
 #include "en_port.h"
 
+static inline struct delayed_work *to_delayed_work(struct work_struct *work)
+{
+	return container_of(work, struct delayed_work, work);
+}
 
 static void mlx4_en_vlan_rx_register(struct net_device *dev, struct vlan_group *grp)
 {
